parquet-hadoop/src/main/java/org/apache/parquet/format/converter/ParquetMetadataConverter.java

```diff
@@ -983,23 +1082,39 @@ public void writeDataPageV2Header(
             valueCount, nullCount, rowCount,
             statistics,
             dataEncoding,
-            rlByteLength, dlByteLength), to);
+            rlByteLength, dlByteLength, null), to);
+  }
+
+  public void writeDataPageV2Header(
+      int uncompressedSize, int compressedSize,
+      int valueCount, int nullCount, int rowCount,
+      org.apache.parquet.column.statistics.Statistics statistics,
+      org.apache.parquet.column.Encoding dataEncoding,
+      int rlByteLength, int dlByteLength,
+      OutputStream to, PrimitiveType type) throws IOException {
+    writePageHeader(
+        newDataPageV2Header(
+            uncompressedSize, compressedSize,
+            valueCount, nullCount, rowCount,
+            statistics,
+            dataEncoding,
+            rlByteLength, dlByteLength, type), to);
   }
 
   private PageHeader newDataPageV2Header(
       int uncompressedSize, int compressedSize,
       int valueCount, int nullCount, int rowCount,
       org.apache.parquet.column.statistics.Statistics<?> statistics,
       org.apache.parquet.column.Encoding dataEncoding,
-      int rlByteLength, int dlByteLength) {
+      int rlByteLength, int dlByteLength, PrimitiveType type) {
     // TODO: pageHeader.crc = ...;
     DataPageHeaderV2 dataPageHeaderV2 = new DataPageHeaderV2(
         valueCount, nullCount, rowCount,
         getEncoding(dataEncoding),
         dlByteLength, rlByteLength);
     if (!statistics.isEmpty()) {
       dataPageHeaderV2.setStatistics(
-          toParquetStatistics(statistics));
+          toParquetStatistics(statistics, type));
```

Comment:
I don't think this is necessary. The type is passed to statistics when it is created so this shouldn't need to pass them together.

Reply:
OK, I'll add the type to the statistics and undo these changes. It'll help with the toString in statistics anyway.